src/julia_gc.c

```diff
@@ -575,7 +636,7 @@ static void GapRootScanner(int full)
 {
     // mark our Julia module (this contains references to our custom data
     // types, which thus also will not be collected prematurely)
-    JMark(Module);
+    jl_gc_mark_queue_obj(JuliaTLS, (jl_value_t *) Module);
```

Comment:
Please add a brief comment here which explains why this does not call `JMark` resp. `MarkJuliaObj` (probably something like this: 
```suggestion
    // We call jl_gc_mark_queue_obj directly here instead of one of our wrappers
    // because we know that <Module> is a valid Julia object here; so we can and
    // should bypass any validation checks.
    jl_gc_mark_queue_obj(JuliaTLS, (jl_value_t *)Module);
```
Although this does not yet explain why `YoungRef` is not relevant here, which should also be stated.